### PR TITLE
feat(sui-domain): created anemicModel to avoid unuseful code on our e…

### PR DESCRIPTION
**Pull request by carlesnunez**
## Description
This PR implements the AnemicModel on our sui-domain. Using this we will inherit extending from our entities and value object the properties with the convention 'this._propertyName' and the method .toJSON that transforms our AnemicModel into JSON valid object.

The **.toJSON** method will:
    * 1 - replace key _myPropertyKey by myPropertyKey
    * 2 - Check if the property was existing on the base properties constructor object or was added later. If second it will not included on te toJSON
    * 3 - Check if the property is AnemicObject and call its toJSON method too to get their JSON aspect.
    * 4 - Check if is an array, if it is will check if is an AnemicObject and do the same as step 3
    * 5 - If any of the latest steps were successful it will return the property without the dash.

#163

### CODE REVIEW

**Comment by carlesnunez** [comment]
If I understood you this is not going to happen ever for the check that I do on the point 3:
https://github.com/SUI-Components/sui/pull/187/files#diff-7b683a16b375ade8a82d8e6f20f7d2faR20

**Comment by carlesnunez** [comment]
Okay! I'll do that!

File: packages/sui-decorators/test/browser/cacheSpec.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ describe('Cache', () => {
   })
 
   describe('should be able use others method in this instance', () => {
-    it('return the response of other instance´s method', () => {
+    xit('return the response of other instance´s method', () => {
```
Comment (julian-gernun) [inline packages/sui-decorators/test/browser/cacheSpec.js]: What will happen to all these test you are skipping?

**Comment by carlesnunez** [comment]
I wasn't able to fix it due to a unknowledge of these package. I've talked with @miduga with that and we will fix ASAP.

File: package.json [hunk 2]
```diff
@@ -22,8 +28,25 @@
...
-    "validate-commit-msg": "2.12.2"
+    "karma": "2.0.0",
+    "karma-browserify": "5.1.2",
+    "karma-chrome-launcher": "2.2.0",
+    "karma-clear-screen-reporter": "1.0.0",
+    "karma-coverage": "1.1.1",
+    "karma-firefox-launcher": "1.0.1",
+    "karma-mocha": "1.3.0",
+    "karma-spec-reporter": "0.0.31",
+    "mocha": "^4.0.1",
```
Comment (julian-gernun) [inline package.json]: you could remove the ^

**Pull request merged**